### ISSUE TITLE
CA2000 fires incorrectly when using `ConfigureAwait(false)`

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#6881]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: 8.0.0-preview.7.23375.6

### Steps To Reproduce

In the following code:
```c#
var fileStream = File.OpenRead("");
await using (fileStream.ConfigureAwait(false))
{
    fileStream.Read(Array.Empty<byte>());
}
```
I get a warning on `File.OpenRead` telling me to call `Dispose`, even though I'm properly disposing it via the using statement. The only fix is to remove the `.ConfigureAwait(false)` - then the warning disappears.

**Comment by mavasani** [comment]
This was fixed recently with #6787.

Verified that this repros on the latest public preview .NET 8 SDK, but does not reproduce on the latest NetAnalyzers NuGet package: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/8.0.0-preview.23425.3
